### ISSUE TITLE
AWS Kubeflow : Pipeline fail - No healthy upstream

### ISSUE THREAD

**Issue by ajinkya933** [kubeflow/pipelines#4730]
### What steps did you take:

I am following this documentation: https://www.kubeflow.org/docs/aws/deploy/install-kubeflow/
Using this tutorial I successfully installed Kubeflow on EKS
After I installed I did:

```
kubectl port-forward svc/istio-ingressgateway -n istio-system 8080:80
Forwarding from 127.0.0.1:8080 -> 80
Forwarding from [::1]:8080 -> 80
```

This starts kubeflow on local host `127.0.0.1:8080` . But when I try to upload a new `pipeline.tar.gz ` file I get error: 
```
pipeline version creation failed 
```

see below screen shot for reference

https://github.com/ajinkya933/Assignment_solution/blob/master/pipelines.png

yaml file used for deployment:

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: kubeflow-cluster
  region: us-east-2
nodeGroups:
  - name: kubenodegroup
    labels: { role: workers }
    instanceType: t3.xlarge
    desiredCapacity: 1
    volumeSize: 80
```

### What happened:
pipeline version creation failed  error
### What did you expect to happen:
I expected the pipeline to execute normally

/kind bug

**Comment by PatrickXYS** [comment]
If you are following latest AWS installation, you don't need `port-forward`, but `kubectl get ingress -n istio-system`.

Given this message `pipeline version creation failed `, it's possible that the kfp related pods are not healthy,

can you run `kubectl get pods -n kubeflow` and paste the output?

**Comment by ajinkya933** [comment]
I tried 
```
$ kubectl get ingress -n istio-system
NAME            HOSTS   ADDRESS   PORTS   AGE
istio-ingress   *                 80      41s
```
but I don't get any IP address. Thats why I am doing `kubectl port-forward svc/istio-ingressgateway -n istio-system 8080:80`

I noticed why this error is coming up its due to that single t3 instance. Now I have a new cluster.yaml file with 2 r4.xlarge instances and this error is nor showing up. but even for this new yaml file I get the same error:

```
$ kubectl get ingress -n istio-system
NAME            HOSTS   ADDRESS   PORTS   AGE
istio-ingress   *                 80      41s
```
so I am doing `kubectl port-forward svc/istio-ingressgateway -n istio-system 8080:80` to start Kubeflow. There are errors in new pipeline which I will track on another issue
